src/lib/Syntax.hs

```diff
@@ -321,6 +318,8 @@ data PrimOp e =
       | Inject e
       | PtrOffset e e
       | PtrLoad e
+      | GetPtr e
+      | MakePtrType e
```

Comment:
Couldn't this be part of the untyped language? We don't even emit it after that, but putting it in here means that we have to handle it everywhere in the compiler